c7n/resources/directory.py

```diff
@@ -50,6 +51,95 @@ class DirectoryVpcFilter(VpcFilter):
     RelatedIdsExpression = "VpcSettings.VpcId"
 
 
+@Directory.filter_registry.register('ldap')
+class DirectoryLDAPFilter(Filter):
+    """Filter directories based on their LDAP status
+
+    :example:
+
+        .. code-block:: yaml
+
+            policies:
+              - name: ldap-enabled-directories
+                resource: directory
+                filters:
+                  - type: ldap
+                    status: Disabled
+    """
+    schema = type_schema(
+        'ldap',
+        status={'type': 'string', 'enum': ['Enabled', 'Disabled']},
+        required=['status']
+    )
+
+    permissions = ('ds:DescribeLDAPSSettings',)
+    annotation_key = 'c7n:LDAPSSettings'
+    valid_directory_types = ['MicrosoftAD']
+
+    def process(self, resources, event=None):
+        resources = self.filter_resources(resources, 'Type', self.valid_directory_types)
+        client = local_session(self.manager.session_factory).client('ds')
+        status = self.data.get('status', 'Enabled')
+        matches = []
+        for r in resources:
+            if self.annotation_key not in r:
+                try:
+                    ldap_settings = client.describe_ldaps_settings(
+                        DirectoryId=r['DirectoryId'])['LDAPSSettingsInfo']
+                except client.exceptions.ClientError:
```

Comment:
I'm not sure `client.exceptions.ClientError` is a valid error type here. Is it mentioned in the docs anywhere?

Reply:
Removed. Thanks!

Reply:
As per our chat, the exception is not needed and is taken out. ClientError exception would only bubble up for directories that are not supported for the API. Those directory types are filtered out here